rflx/parser/parser.py

```diff
@@ -153,6 +153,13 @@ def check_types(types: Mapping[ID, Type]) -> None:
                 + ", ".join(sorted(identical_literals))
             )
 
+    literals = {l: t for t in types.values() if isinstance(t, Enumeration) for l in t.literals}
+    name_conflicts = set(literals) & set(str(t.name) for t in types)
+    for name in sorted(name_conflicts):
+        raise ParserError(
+            f'literal in enumeration "{literals[name].identifier}" conflicts with type "{name}"'
+        )
+
```

Comment:
This will always raise an exception in the first loop iteration. Either we should use the first element or (preferably) concatenate all conflicts into a single error message.

Reply:
I know, I also thought about replacing the for-loop by something like:

```Python
if name_conflicts:
    name = sorted(name_conflicts)[0]
```

But I see no advantage in this variant. I also thought about changing the error message to list all affected entities, but didn't find a good way to do that (just concatenating the current message would be quite ugly).

I think we should generally support multiple errors like proposed in https://github.com/Componolit/RecordFlux/issues/248#issuecomment-630716961. I would suggest leaving it as it is and improve it in #248.

Reply:
Fair enough. The loop works well with the interface I have in mind for #248. Let's keep it as is then.